lib/db/DB.js

```diff
@@ -16,19 +19,61 @@ class DB {
   }
 
   define() {
-    this.Order = this.sequelize.define('order', {
-      id: { type: Sequelize.INTEGER, primaryKey: true, autoIncrement: true },
-      peerId: Sequelize.INTEGER,
-      invoice: Sequelize.STRING,
-      quantity: Sequelize.DECIMAL(14, 8),
-      price: Sequelize.DECIMAL(9, 8),
+    // P2P
+
+    this.Peer = this.sequelize.define('peers', {
+      id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
+      nodeKey: { type: DataTypes.STRING, unique: true },
+      ipv4: DataTypes.STRING(15),
+      port: DataTypes.SMALLINT,
     });
 
-    this.Peer = this.sequelize.define('peer', {
-      id: { type: Sequelize.INTEGER, primaryKey: true, autoIncrement: true },
-      nodeKey: { type: Sequelize.STRING, unique: true },
-      ipv4: Sequelize.STRING(15),
-      port: Sequelize.SMALLINT,
+    // DOB
+
+    this.Currency = this.sequelize.define('currencies', {
+      id: { type: DataTypes.STRING, primaryKey: true },
+    });
+
+    //
+
+    this.Pair = this.sequelize.define('pairs', {
+      id: { type: DataTypes.STRING, primaryKey: true },
```

Comment:
What are your thoughts here regarding making `id` a string? Would an example id be like `BTCLTC`? I'm thinking we can just make this an auto-incrementing `int` and get the currency pairs from `baseCurrency` and `quoteCurrency`.

Reply:
`currencies` and `pairs` should be recognized globally (p2p network), and not by when they were added to a local node.  using the currency string symbol might be the easiest way to do so.

Reply:
Makes perfect sense for `currencies` having the ticker string be the only identifier, and I see what you mean for `pairs`. After thinking about this a bit, I think the best approach is probably to have `baseCurrency` and `quoteCurrency` act as a composite primary key. That's the only data we really need to track, and it'll make it simpler to perform queries like "select all orders with BTC as base currency".

See [here](https://gist.github.com/lucasscariot/5b8747fbc8a6948a805c646fae4ceef8) for an example of a composite primary key in sequelize.